Resources/Storage/GFAL2_StorageBase.py

```diff
@@ -1229,24 +1231,6 @@ def __listSingleDirectory( self, path ):
     try:
       listing = self.gfal2.listdir( path )
 
-      files = {}
-      subDirs = {}
-
-      for entry in listing:
-        fullPath = '/'.join( [ path, entry ] )
-        self.log.debug( 'GFAL2_StorageBase.__listSingleDirectory: path: %s' % fullPath )
-        res = self.__getSingleMetadata( fullPath )
-        if res['OK']:
-          metadataDict = res['Value']
-          if metadataDict['Directory']:
-            subDirs[fullPath] = metadataDict
-          elif metadataDict['File']:
-            files[fullPath] = metadataDict
-          else:
-            self.log.debug( "GFAL2_StorageBase.__listSingleDirectory: found item which is neither file nor directory", fullPath )
-
-      return S_OK( {'SubDirs' : subDirs, 'Files' : files} )
-
     except gfal2.GError, e:
       if e.code == errno.ENOENT:
```

Comment:
I'm not sure what you mean, for me there is also an else: clause that would catch other exceptions?